src/3d/symbols/qgspolygon3dsymbol_p.cpp

```diff
@@ -98,11 +98,33 @@ bool QgsPolygon3DSymbolHandler::prepare( const Qgs3DRenderContext &context, QSet
 
   const QgsPhongTexturedMaterialSettings *texturedMaterialSettings = dynamic_cast<const QgsPhongTexturedMaterialSettings *>( mSymbol->materialSettings() );
 
-  outNormal.tessellator.reset( new QgsTessellator( chunkOrigin.x(), chunkOrigin.y(), true, mSymbol->invertNormals(), mSymbol->addBackFaces(), false, texturedMaterialSettings && texturedMaterialSettings->requiresTextureCoordinates(), mSymbol->renderedFacade(), texturedMaterialSettings ? texturedMaterialSettings->textureRotation() : 0 ) );
-  outSelected.tessellator.reset( new QgsTessellator( chunkOrigin.x(), chunkOrigin.y(), true, mSymbol->invertNormals(), mSymbol->addBackFaces(), false, texturedMaterialSettings && texturedMaterialSettings->requiresTextureCoordinates(), mSymbol->renderedFacade(), texturedMaterialSettings ? texturedMaterialSettings->textureRotation() : 0 ) );
-
-  outNormal.tessellator->setOutputZUp( true );
-  outSelected.tessellator->setOutputZUp( true );
+  QgsTessellator *tessellator = new QgsTessellator();
+  tessellator->setOrigin( chunkOrigin );
+  tessellator->setAddNormals( true );
+  tessellator->setInvertNormals( mSymbol->invertNormals() );
+  tessellator->setAddBackFaces( mSymbol->addBackFaces() );
+  tessellator->setExtrusionEnabled( true );
+  tessellator->setOutputZUp( true );
+  tessellator->setExtrusionFaces( mSymbol->extrusionFaces() );
+  tessellator->setTextureRotation( texturedMaterialSettings ? static_cast<float>( texturedMaterialSettings->textureRotation() ) : 0 );
```

Comment:
clang-tidy might cry (also for the outNormal tessellator below)
```suggestion
  tessellator->setTextureRotation( texturedMaterialSettings ? static_cast<float>( texturedMaterialSettings->textureRotation() ) : 0.f );
```